File: jadx-gui/src/main/java/jadx/gui/treemodel/JInputScripts.java
package jadx.gui.treemodel;

import java.nio.file.Path;
import java.util.List;

import javax.swing.Icon;
import javax.swing.ImageIcon;
import javax.swing.JPopupMenu;

import jadx.gui.ui.MainWindow;
import jadx.gui.utils.NLS;
import jadx.gui.utils.UiUtils;
import jadx.gui.utils.ui.SimpleMenuItem;

public class JInputScripts extends JNode {
	private static final ImageIcon INPUT_SCRIPTS_ICON = UiUtils.openSvgIcon("nodes/scriptsModel");

	public JInputScripts(List<Path> scripts) {
		for (Path script : scripts) {
			add(new JInputScript(script));
		}
	}

	@Override
	public JPopupMenu onTreePopupMenu(MainWindow mainWindow) {
		JPopupMenu menu = new JPopupMenu();
		menu.add(new SimpleMenuItem(NLS.str("popup.add_scripts"), mainWindow::addFiles));
		menu.add(new SimpleMenuItem(NLS.str("popup.new_script"), mainWindow::addNewScript));
		return menu;
	}

	@Override
	public JClass getJParent() {
		return null;
	}

	@Override
	public Icon getIcon() {
		return INPUT_SCRIPTS_ICON;
	}

	@Override
	public String makeString() {
		return NLS.str("tree.input_scripts");
	}
}

File: jadx-plugins/jadx-script-kotlin/src/main/kotlin/jadx/plugins/script/kotlin/gui/JInputScripts.kt
package jadx.plugins.script.kotlin.gui

import jadx.api.plugins.JadxPluginContext
import jadx.gui.treemodel.JClass
import jadx.gui.treemodel.JNode
import jadx.gui.ui.MainWindow
import jadx.gui.utils.NLS
import jadx.gui.utils.UiUtils
import jadx.gui.utils.ui.SimpleMenuItem
import java.nio.file.Path
import javax.swing.Icon
import javax.swing.ImageIcon
import javax.swing.JPopupMenu

class JInputScripts(
	pluginContext: JadxPluginContext,
	scripts: List<Path>,
) : JNode() {
	companion object {
		private val INPUT_SCRIPTS_ICON: ImageIcon = UiUtils.openSvgIcon("nodes/scriptsModel")
	}

	init {
		for (script in scripts) {
			add(JInputScript(pluginContext, script))
		}
	}

	override fun onTreePopupMenu(mainWindow: MainWindow): JPopupMenu {
		val menu = JPopupMenu()
		menu.add(SimpleMenuItem(NLS.str("popup.add_scripts")) { mainWindow.addFiles() })
		menu.add(SimpleMenuItem(NLS.str("popup.new_script")) { mainWindow.addNewScript() })
		return menu
	}

	override fun getJParent(): JClass? {
		return null
	}

	override fun getIcon(): Icon {
		return INPUT_SCRIPTS_ICON
	}

	override fun getID(): String {
		return "JInputScripts"
	}

	override fun makeString(): String {
		return NLS.str("tree.input_scripts")
	}
}
